addons/mail/controllers/discuss/channel.py

```diff
@@ -157,7 +157,10 @@ def discuss_channel_sub_channel_create(self, parent_channel_id, from_message_id=
         if not channel:
             raise NotFound()
         sub_channel = channel._create_sub_channel(from_message_id, name)
-        return Store(sub_channel).add(sub_channel, {"forceOpen": True}).get_result()
+        return {
+            "data": Store(sub_channel).add(sub_channel).get_result(),
+            "sub_channel": Store.one_id(sub_channel),
```

Comment:
`one_id()` has the advantage to require less maintenance, because we don't spread the shape of identifying data of sub channel everywhere in the code.

I get that `one_id()` is so-so, but current alternatives for not using it is worse. I would not rush in stating something is deprecated when it works well for the use and is easily maintainable like these LOCs.

Also future plans to improve `one_id()` can change these LOCs. As I said, this doesn't cost much. Forcing using alternatives now just makes code less maintainable...